### PR TITLE
Fixes monodocs by updating intersphinx mapping

### PR DESCRIPTION
**Pull request by thomasjpfan**
## Why are the changes needed?

The monodocs are failing because PyTorch has moved to using `main` as their development branch.

## What changes were proposed in this pull request?

This PR fixes the `intersphinx_mapping`.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -287,14 +287,13 @@
...
     "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
-    "torch": ("https://pytorch.org/docs/master/", None),
+    "torch": ("https://pytorch.org/docs/main/", None),
     "scipy": ("https://docs.scipy.org/doc/scipy/reference", None),
     "matplotlib": ("https://matplotlib.org", None),
     "pandera": ("https://pandera.readthedocs.io/en/stable/", None),
 }
 
 myst_enable_extensions = ["colon_fence"]
-myst_heading_anchors = 6
```
Comment (neverett) [inline docs/conf.py:297]: This should not be deleted -- this is the heading level depth to assign HTML anchors. (The default is zero, so if removed, I think we lose HTML anchors.)

**Review by neverett — CHANGES_REQUESTED** [review]
One change requested, otherwise LGTM

**Comment by neverett** [comment]
Looks like this was fixed in https://github.com/flyteorg/flyte/pull/5755

File: docs/conf.py (hunk 1 above)
Comment (thomasjpfan) [reply docs/conf.py]: Ah sorry, this was a mistake. Reverted.

**Comment by thomasjpfan** [comment]
Looks like this was already fix on master. Closing.

**Pull request closed without merging**